### DISCUSSION TITLE
Documentation on extensions possibilities

### DISCUSSION THREAD

**Discussion by PierreLouisEx9** [foxglove/community#810] [Q&A] [answered]
Hello everyone,

I'm beginning to look into the extension creation for improving Foxglove data visualization. But i can't find anywhere some sort of documentation where i can see which types or functions are available in the @foxglove/studio. The closest thing is the tutorials for creating a message converter or a custom panel (which are really poorly documented tbh), but i see a lot of different libraries and still no information in sight.

If anybody has some tips to help me find what i'm searching, i'll be grateful.

Thanks a lot

**Comment by achim-k** [comment, marked as answer]
You can find the docs for extensions here: https://foxglove.dev/docs/studio/extensions/getting-started
The types of `@foxglove/studio` can be found in https://github.com/foxglove/studio/tree/main/packages/studio (referenced by the documentation)

**Reply by PierreLouisEx9** [reply]
Thank you for your quick answer,

As i understand, i need to become a JavaScript developer if i want to freely custom my foxglove app. I think i'll stick to the basic app and just create ros2 nodes to do the transformations that i need on the data.

Best regards
